### PR TITLE
Add lead and lag to java [skip ci]

### PR DESCRIPTION
**Pull request by revans2**
This is based off of #6277 and will stay a WIP until it is merged in.

This adds in java APIs for the lead and lag functionality done as a part of the PR.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by mythrocks — APPROVED** [review]
Thanks for this change, @revans2. :+1: 

We might need to rebase, given that some changes were requested in #6277 during the review.

**Comment by revans2** [comment]
@mythrocks I upmerged.  Will do it again once your PR goes in.  @jlowe I would appreciate a review of the java/JNI code early so we can merge this in ASAP after the dependency is merged in.

File: java/src/main/java/ai/rapids/cudf/Aggregation.java
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/Aggregation.java:52]: Are these the correct IDs?  My counting of enums in aggregation.hpp ends up with these being 20 and 21, respectively.  The Java list seems off starting after `COUNT` because the C++ header has two values instead, `COUNT_VALID` and `COUNT_ALL`.

I guess it works because the JNI code doesn't use the `enum Kind` from C++ but hardcodes the values, but it is quite confusing to follow since it's sorta trying to follow that enum without actually being that enum.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java:2873]: Nit: would be nice to move `viewHandle` argument to its own line to be consistent with the arg-per-line of the other args.  This also makes it consistent with the declaration of `Table.rollingWindowAggregate`.

File: java/src/main/java/ai/rapids/cudf/Aggregation.java
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/Aggregation.java:279]: Comment says it will get a `ColumnVector` but it returns a long.

**Review by jlowe — APPROVED** [review]

**Pull request merged**
